crates/rpc-client/src/utils.rs

```diff
@@ -40,9 +41,94 @@ impl TracingHttpClient {
             Span::current().record("params", field::display(&params));
         }
 
-        self.inner
-            .rpc(method, params)
-            .await
-            .with_context(|| format!("rpc {method}"))
+        let mut backoff = ExponentialBackoff::new(Duration::from_secs(1));
+
+        loop {
+            match self.inner.rpc(method, params).await {
+                Ok(r) => return Ok(r),
+                Err(e) => {
+                    // Retry on reqwest errors. CKB RPCs are almost all safe to retry.
+                    if e.is::<reqwest::Error>() {
+                        let next = backoff.next_sleep();
+                        // reqwest::Error displays the whole chain, no need to use {:#}.
+                        tracing::warn!(
+                            "rpc client error, will retry in {:.2}s: {}",
+                            next.as_secs_f64(),
+                            e,
+                        );
+                        tokio::time::sleep(next).await;
+                        continue;
+                    }
+                    return Err(e.context(format!("rpc {method}")));
+                }
+            }
+        }
+    }
+}
+
+pub struct ExponentialBackoff {
```

Comment:
Had to move this to avoid a circular dependency between `gw-utils` and `gw-rpc-client`.